boards/cc2538dk/Makefile.include

```diff
@@ -24,7 +24,7 @@ export PROGRAMMER ?= cc2538-bsl
 
 ifeq ($(PROGRAMMER),cc2538-bsl)
   FLASHER = $(RIOTTOOLS)/cc2538-bsl/cc2538-bsl.py
-  FFLAGS  = -p "$(PORT)" -e -w -v $(FLASHFILE)
+  FFLAGS  = -p "$(PORT)" -a $(IMAGE_OFFSET)+$(ROM_START_ADDR) -w -v $(FLASHFILE)
```

Comment:
This should still be edited according to @cladmi suggestion, I would maybe split it in something like:

```
  FFLAGS  = -p "$(PORT)" -w -v $(FLASHFILE)
  FFLAGS += $(if $(IMAGE_OFFSET), -a $(shell printf "0x%08x" $$(($(IMAGE_OFFSET) + $(ROM_START_ADDR)))))
```

Reply:
[@fjmolinas] This solution is looking good. I tested **riotboot/flash-extended-slot0**, and **riotboot/flash-combined-slot0**. Both works :)
But we currently have to flash the bootloader first (riotboot/flash-bootloader), and then the firmware. How do you suggest we integrate the 2, so we only need to use 1 command (i.e. riotboot/flash-slot0)

Reply:
> But we currently have to flash the bootloader first (riotboot/flash-bootloader), and then the firmware. How do you suggest we integrate the 2, so we only need to use 1 command (i.e. riotboot/flash-slot0)

why? Because of the cca field?

Reply:
I tested, by flashing the hello-world example, then riotboot/flashXXXX.
riotboot/flash-extended-slot0 and riotboot/flash-combined-slot0 works fine,
However, riotboot/flash-slotX fails unless if I use riotboot/flash-bootloader first.
I believe it is because of the CCA entry point as you mentioned previously.

Reply:
To program the CC2538 on the smartrf06eb, I currently have to hold the select button whilst pressing the reset button, before I flash it. Is there any way for me to bypass this procedure?

Reply:
> However, riotboot/flash-slotX fails unless if I use riotboot/flash-bootloader first

Hmm. Maybe the filter I defined isn't correct. What I wanted to achieve was that `cca` never be updated for any slot and when compiling `riotboot/flash-extended-slot0 and riotboot/flash-combined-slot0` it shouldn't either. Maybe look at the bin files and see it isn't defined? If not, changing it to:

```
# The entry point `cortex_vector_base` is defined in the cca field.
# If the cca field is updated when flashing a slot then the entry 
# point will never be the bootloader but the respective slot. This
# ensures it never happens .
ifneq (,$(filter riotboot,$(FEATURES_REQUIRED)))
  CFLAGS+=-DUPDATE_CCA=0
endif
```
would work as long as you flash some RIOT application before, like hello-world. But I'm not sure of this approach. The ideal would be that when doing  `riotboot/flash-extended-slot0`,  `riotboot/flash-combined-slot0` or `riotboot/flash-bootloader` the cca is updated and points to the bootloader, but in no other case. I'll try to figure something out. but I don't have that much time right now, soy I might take a while to respond :) (I'll be AFK most of next week too just so you now).

I would suggest you look into ways of using `UPDATE_CCA` flags to achieve what I'm suggesting :)

Reply:
Thanks a lot for the help. I will continue to work on it :)

Reply:
I prefer a second variable over adding to `FFLAGS` as it prevents from overwriting it from outside. I am slowly trying to remove the `FFLAGS +=`.

For the `cca`, can we not include it in the linker if there is a `rom_offset` instead of a compile time hack ?
The hack does not work, as the bootloader uses `riotboot`.
General note, checking `FEATURES_REQUIRED` is forbidden (there is a sanity check for it) and should be done with `FEATURES_USED`.

Somehow I would still like that the `combined-extended` firmwares work because it is what should be distributed and is required for the tests to pass. But I do not know how to handle it properly yet.